astropy/modeling/core.py

```diff
@@ -2462,8 +2460,16 @@ def render_model(model, arr=None, coords=None):
             assert coords[0].shape == arr.shape
         else:
             arr = np.zeros(coords[0].shape)
+    
+    if arr is not None:
+        arr = arr.copy()
+        # Check dimensions match model
+        assert arr.ndim == model.n_inputs
 
```

Comment:
These blocks got swapped during a bug hunt. It doesn't really matter except that there's no reason to copy the array if it's just been made from the coordinates, so I'll swap it back.